ecs-agent/daemonimages/csidriver/util/utils_windows.go

```diff
@@ -22,10 +22,56 @@ limitations under the License.
 
 package util
 
-import "errors"
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"unsafe"
 
-// IsBlockDevice checks if the given path is a block device
+	"golang.org/x/sys/windows"
+)
+
+// Define functions from Windows API
+var (
+	// We will define them here so that they are loaded by default for Windows
+	// instead of loading them on each invocation of this method.
+	kernel32                              = windows.NewLazySystemDLL("kernel32.dll")
+	procGetVolumeNameForVolumeMountPointW = kernel32.NewProc("GetVolumeNameForVolumeMountPointW")
+
+	// funcGetVolumeNameForVolumeMountPointW is the GetVolumeNameForVolumeMountPointW Win32 API.
+	// Reference: https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getvolumenameforvolumemountpointw
+	funcGetVolumeNameForVolumeMountPointW func(a ...uintptr) (r1 uintptr, r2 uintptr, lastErr error) = procGetVolumeNameForVolumeMountPointW.Call
+)
+
+// IsBlockDevice checks if the given path is a block device on Windows.
 func IsBlockDevice(fullPath string) (bool, error) {
-	// TODO
-	return false, errors.New("not supported")
+	// Ensure the fullPath ends with a backslash.
+	// This is as per the API documentation- https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getvolumenameforvolumemountpointw#parameters
+	if !strings.HasSuffix(fullPath, "\\") {
```

Comment:
Q: From the doc, it seems like only one `\` is needed (for example, "Y:\MountX\"). Any reason why we're appending 2?

Reply:
The second `\` is for escaping purposes. The string when read in Go would have escaped `\` and therefore, we need to cater to the same.